File: Homeworks/C# Part 2/06.StringsAndTextProcessing/12.ParseURL/ParseURL.cs
using System;

class ParseURL
{
    static void Main()
    {
        string input = Console.ReadLine();
        int firstSeparatorIndex = input.IndexOf("://");
        string protocol = input.Remove(firstSeparatorIndex);
        input = input.Remove(0, firstSeparatorIndex + 3);
        int secondSeparatorIndex = input.IndexOf("/");
        string server = input.Remove(secondSeparatorIndex);
        string resource = input.Remove(0, secondSeparatorIndex);
        Console.WriteLine("[protocol] = {0}", protocol);
        Console.WriteLine("[server] = {0}", server);
        Console.WriteLine("[resource] = {0}", resource);
    }
}

File: Homeworks/JS Fundamentals/11. Strings/07. Parse URL/parse-url.js
function solve(args) { 
    var firstSeparatorIndex = args[0].indexOf('://');
    var secondSeparatorIndex = args[0].indexOf('/', firstSeparatorIndex + 3);
    var protocol = args[0].substring(0, firstSeparatorIndex);    
    var server = args[0].substring(firstSeparatorIndex + 3, secondSeparatorIndex);
    var resource = args[0].substring(secondSeparatorIndex);
    console.log('protocol: ' + protocol);
    console.log('server: ' + server);
    console.log('resource: ' + resource);
}